### PR TITLE
HTML-598 : Add Boolean datatype support to the <when></when> markup

### PR DESCRIPTION
**Pull request by rbuisson**
HTML-598 : Pull request to add the Boolean datatype support to the markup.
https://issues.openmrs.org/browse/HTML-598

### CODE REVIEW

File: api/src/main/java/org/openmrs/module/htmlformentry/handler/WhenTagHandler.java [hunk 1]
```diff
@@ -38,17 +38,26 @@ protected String getSubstitution(FormEntrySession session, FormSubmissionControl
...
+			obs.whenValueThenDisplaySection(whenCondition, thenDisplay);
+		}
+		if (thenJavaScript != null) {
+			obs.whenValueThenJavaScript(whenCondition, thenJavaScript);
+		}
+		if (elseJavaScript != null) {
+			obs.whenValueElseJavaScript(whenCondition, elseJavaScript);
+		}
         return "";
     }
```
Comment (mogoodrich) [inline api/src/main/java/org/openmrs/module/htmlformentry/handler/WhenTagHandler.java]: Looks good.  Can you add a quick unit test as well? It wasn't obvious where the existing test for this tag was, but looks like it is in ObsTagTest:testObsWithControlsSection.  This is just a test that on the server-side the correct javascript is generated. Thanks!

**Comment by rbuisson** [comment]
OK, thanks for the tip about this JUnit test. I would have never find it else.
Junit test ObsTagTest:testObsWithControlsSection is now updated.
(Hope I did it the right way)

File: api/src/main/java/org/openmrs/module/htmlformentry/handler/WhenTagHandler.java (hunk 1 above)
Comment (rbuisson) [inline api/src/main/java/org/openmrs/module/htmlformentry/handler/WhenTagHandler.java]: See next commit

**Comment by dkayiwa** [comment]
@mogoodrich do you think this is ready for merging? :)

**Comment by mogoodrich** [comment]
Thanks for the reminder Daniel... I will merge this in.

**Pull request merged**